tests/database-dispatcher-test.js

```diff
@@ -0,0 +1,174 @@
+		mock(path.join(process.cwd(), 'config', 'database.json'), {
+			core: {
+				type: 'mysql',
+				host: 'foo',
+				user: 'root',
+				password: 'foobar',
+				database: 'my_db',
+				port: '1234'
+			},
+
+			foo: {
+				type: 'mongodb',
+				host: 'foo',
+				user: 'root',
+				password: 'foobar',
+				database: 'my_db',
+				port: '1234'
+			},
+
+			_default: {
+				type: 'mysql',
+				host: 'foo',
+				user: 'root',
+				password: 'foobar',
+				database: 'my_db',
+				port: '1234'
+			}
+		});
+	};
+
+	const databaseMock = () => {
+		mock(path.join(process.cwd(), 'node_modules', '@janiscommerce/mysql'), './../mocks/database-mock');
+		mock(path.join(process.cwd(), 'node_modules', '@janiscommerce/mongodb'), './../mocks/database-mock');
+	};
+
+	beforeEach(() => {
+		configMock();
+		databaseMock();
+	});
+
+	afterEach(() => {
+		mock.stopAll();
+	});
+
+	describe('getters', function() {
+
+		it('should return db types', function() {
+			assert.equal(typeof DatabaseDispatcher.dbTypes, 'object');
+		});
+
+
+		it('should return database config path', function() {
+			assert.equal(typeof DatabaseDispatcher.configPath, 'string');
+		});
+
+
+		it('should return database config object', function() {
+			assert.equal(typeof databaseDispatcher.databaseConfig, 'object');
+		});
+	});
+
+	describe('getDBDriver', function() {
+
+		it('should return MySQL module', function() {
+			assert.equal(typeof DatabaseDispatcher.getDBDriver({ type: 'mysql' }), 'function');
+		});
+
+		it('should return MongoDB module', function() {
+			assert.equal(typeof DatabaseDispatcher.getDBDriver({ type: 'mongodb' }), 'function');
+		});
+	});
+
+	describe('getDatabase', function() {
+
+		it('should return database connection (MySQL)', function() {
+			assert.equal(typeof databaseDispatcher.getDatabase('core'), 'object');
+		});
+
+		it('should return database connection (MongoDB)', function() {
+			assert.equal(typeof databaseDispatcher.getDatabase('foo'), 'object');
+		});
+
+		it('should return database connection (Default)', function() {
+			assert.equal(typeof databaseDispatcher.getDatabase(), 'object');
+		});
+	});
+
+	describe('caches', function() {
+
+		it('should return all databases connection object', function() {
+			assert.equal(typeof databaseDispatcher.databases, 'object');
+		});
+
+		it('should return core database connection object', function() {
+			assert.equal(typeof databaseDispatcher.databases.core, 'object');
+		});
+
+		it('should return config object', function() {
+			assert.equal(typeof databaseDispatcher.config, 'object');
+		});
+
+		it('should return database connections', function() {
+
+			assert.equal(typeof databaseDispatcher.getDatabase('core'), 'object');
+			assert.equal(typeof databaseDispatcher.getDatabase('foo'), 'object');
+			assert.equal(typeof databaseDispatcher.getDatabase(), 'object');
+		});
+	});
+
+	describe('clearCaches', function() {
+
+		it('should delete config and database caches', function() {
```

Comment:
Este test no tiene sentido por si solo. No tiene sentido si se corre aislado de los demás tests porque nunca se setea un cache.
Para hacer independiente el test, te recomiendo usar `it.only()` cuando estas desarrollando y probar seteando caches, chequear que tengan contenido, llamar a tu metodo y que ahi no tengan contenido :+1: